libs/wingsdk/src/target-sim/service.ts

```diff
@@ -46,11 +63,59 @@ export class Service extends cloud.Service implements ISimulatorResource {
   }
 
   public onLift(host: IInflightHost, ops: string[]): void {
-    bindSimulatorResource(__filename, this, host);
+    bindSimulatorResource(__filename, this, host, ops);
     super.onLift(host, ops);
   }
 
   public _toInflight(): string {
     return makeSimulatorJsClient(__filename, this);
   }
 }
+
+/** @internal */
+export const SERVICE_HELPER_FQN = fqnForType("sim.ServiceHelper");
+
+/** @internal */
+export interface ServiceHelperProps {
+  readonly service: Service;
+  readonly autoStart: boolean;
+}
+
+/**
+ * This is a helper resource that automatically starts the service after the
+ * service is created in the simulator and shuts it down when the simulator
+ * stops.
+ *
+ * Suppose a service puts an object in a bucket when it starts. The policy
+ * (sim.Policy) that grants the service bucket permissions only takes effect
+ * after the service is created. Because of this, the service needs to be
+ * started after both the service and the policy are created. Vice versa, the
+ * service needs to be stopped before the policy is deleted.
+ *
+ * @internal
+ */
+export class ServiceHelper extends Resource implements ISimulatorResource {
```

Comment:
I feel like this should be avoidable but I can't think of a good way. Do you think requiring this is related to a more fundamental design issue and/or is it just a reasonable tiny bit of indrection that should be expected?

Reply:
Not sure - I suspect we'll learn more once we implement https://github.com/winglang/wing/issues/5305. If we see these helpers become common then it I'd want to revisit this design, or probably just extend it with some way to model inline permissions.